src/amber/router/cookies.cr

```diff
@@ -215,21 +259,20 @@ module Amber::Router
 
       def get(name)
         if value = @store.get(name)
-          deserialize name, decrypt_and_verify(value)
+          verify_and_decrypt(value)
         end
       end
 
       def set(name : String, value : String, path : String = "/", expires : Time? = nil, domain : String? = nil, secure : Bool = false, http_only : Bool = false, extension : String? = nil)
-        cookie = HTTP::Cookie.new(name, value, path, expires, domain, secure, http_only, extension)
-        cookie.value = @encryptor.encrypt_and_sign(serialize(name, {"value": cookie.value}))
+        cookie = HTTP::Cookie.new(name, @encryptor.encrypt_and_sign(value), path, expires, domain, secure, http_only, extension)
         raise Exceptions::CookieOverflow.new if cookie.value.bytesize > MAX_COOKIE_SIZE
         @store[name] = cookie
       end
 
-      private def decrypt_and_verify(encrypted_message)
+      private def verify_and_decrypt(encrypted_message)
         String.new(@encryptor.decrypt_and_verify(encrypted_message))
```

Comment:
kinda weird that the method is called `verify_and_decrypt` and then it calls `@encryptor.decrypt_and_verify`